### PR TITLE
Added some defaults for GlideN64 mupen64plus plugin. Do not delete cf…

### PR DESCRIPTION
**Pull request by dankcushions**
…g every install/update.

see https://github.com/RetroPie/RetroPie-Setup/issues/1511
probably needs some changes as per my comment there  :)

### CODE REVIEW

File: scriptmodules/emulators/mupen64plus.sh [hunk 1]
```diff
@@ -125,19 +125,37 @@ function configure_mupen64plus() {
     cp "$scriptdir/scriptmodules/$md_type/$md_id/mupen64plus.sh" "$md_inst/bin/"
     chmod +x "$md_inst/bin/mupen64plus.sh"
 
-    # to solve startup problems delete old config file
-    rm -f "$md_conf_root/n64/mupen64plus.cfg"
+    # if the user has an existing mupen64plus config we will not overwrite it, but instead copy the
+    # default configuration to mupen64plus.cfg.rp-dist so any new options can be manually
+    # copied across as needed without destroying users changes
```
Comment (joolswills) [inline scriptmodules/emulators/mupen64plus.sh]: please add `local config`

**Comment by joolswills** [comment]
Looks good apart from the missing local config - please add that and I will merge. thanks.

**Comment by joolswills** [comment]
@gizmo98 just want to make sure this is ok as you wrote this part - running mupen64plus creates the initial config, or do we copy it from the intall folder or both ? - just wondering if that will interfere with the rp-dist changes.

**Comment by dankcushions** [comment]
> Looks good apart from the missing local config - please add that and I will merge. thanks.

done :)

**Comment by joolswills** [comment]
@dankcushions thanks. I think there could be another problem, but I've not tested it - just from looking at the code. I believe running the mupen64plus creates an initial config - in which case, the rp-dist wont actually be a default config, but just contain some parts in the case that a default config already exists - please can you check that ?

**Comment by gizmo98** [comment]
Sorry. I cannot test any changes at the moment. I'm on vacation, i have no pi and the internet connection is not the best. Actually i dislike rpi related defaults here. Just ask gonetz to add rpi related defaults here:
https://github.com/gonetz/GLideN64/blob/master/src/Config.cpp
Use `#ifdef VC` (VC = videcore) for raspberry pi stuff. There is an example in the current source code.

**Comment by joolswills** [comment]
that's a good point. and actually yeh if we are adding defaults here we need to do them only for the rpi

**Comment by dankcushions** [comment]
Squashed some commits following some changes suggested by @joolswills. I agree - I think these defaults would be better moved to the GLideN64 repo, but it's moving so fast with RPI-related changes that maybe we should do that at a later stage? I think the ones here so far aren't really that significant - was hoping others would suggest more :)

**Comment by loganmc10** [comment]
I disagree with having the rpi defaults baked into GlideN64, the bilinear mode thing is just a preference. The cache size at 192 is also just a preference, the rpi can handle the default of 500, I have gpu_mem at 600 on my rpi and I use the default 500MB cache

**Comment by joolswills** [comment]
there will always be people who disagree with the defaults - but you can still change them. For now I'm happy with our default config though (which you are free to change). If people want to push some changes upstream, that can be discussed with the author.

**Pull request merged**